subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaPlugin.java

```diff
@@ -372,6 +373,7 @@ private Configuration createRuntimeElements(Project project, final SourceSet mai
                 .withDescription("Elements of runtime for main.")
                 .extendsFrom(implementationConfiguration, runtimeOnlyConfiguration));
         defaultConfiguration.extendsFrom(runtimeElementsConfiguration);
+        ((ConfigurationInternal) runtimeElementsConfiguration).setCanBeDeclaredAgainst(false);
```

Comment:
❔ Why extract this from the builder?

Reply:
This is what caused (I think, they don't have a reproducer) the second error in the bug report. See the original PR which added this without deprecating declaration first: https://github.com/gradle/gradle/pull/21768/files#diff-d50000c56d36151bbac24c48dd8259210337e8707ccf19df58be08ec7905825e

This is used as the foundation for the configurations created by feature variants and test fixtures.

Reply:
Ah, OK.  I should have traced the calls to it, thought it was purely a style thing.